src/script/services/screenshots.ts

```diff
@@ -0,0 +1,54 @@
+import { Icon } from '../utils/interfaces';
+import { getManifest, updateManifest } from './manifest';
+
+const screenshotServiceBaseUrl = 'https://pwa-screenshots.azurewebsites.net';
+
+enum EndPoints {
+  colorScheme = 'getColorScheme',
+  base64 = 'screenshotsAsBase64Strings',
+  zip = 'downloadScreenshotsZipFile',
+}
+
+export async function generateScreenshots(screenshotsList: Array<string>) {
+  try {
+    let screenshots: Array<Icon> = getManifest().screenshots ?? [];
+
+    console.log('generateScreenshots', screenshots);
+
+    const res = await fetch(`${screenshotServiceBaseUrl}/${EndPoints.base64}`, {
+      headers: {
+        'content-type': 'application/json',
+      },
+      method: 'POST',
+      body: JSON.stringify({
+        url: screenshotsList,
+      }),
+    });
+
+    if (res.ok) {
+      screenshots = screenshots.concat(await res.json());
+
+      updateManifest({
+        screenshots,
+      });
+    }
+  } catch (e) {
+    console.error(e);
+  }
+}
+
+export async function downloadScreenshotZip() {
```

Comment:
Just a heads up, there is a download method in the publish page that you can use to base this method off of. Its using this https://github.com/GoogleChromeLabs/browser-fs-access so we can use the File System Access API where available but still support Firefox and Safari with automagic fallbacks haha.